### PR TITLE
[DeadCode] Skip property fetch on cond as may be overridden on RemoveAlwaysTrueIfConditionRector

### LINKED ISSUE #8413: RemoveAlwaysTrueIfConditionRector removes if-statement that is required for undefined array key check
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.19.0             |

The RemoveAlwaysTrueIfConditionRector tries to remove an if-statement that is needed to prevent access to undefined array keys.

## Minimal PHP Code Causing Issue

https://getrector.com/demo/a1bd0aae-6e03-4fe7-b3c5-a6ccbaaff5bc

`
PHP Warning:  Undefined array key "add" in rectorTest.php on line 22
PHP Fatal error:  Uncaught TypeError: array_merge(): Argument #1 must be of type array, null given in rectorTest.php:22
Stack trace:
#0 rectorTest.php(22): array_merge()
#1 rectorTest.php(31): Test->test()
#2 {main}
  thrown in rectorTest.php on line 22
`

## Expected Behaviour

There should be no change as the "if" is required.

### LINKED ISSUE #8583: Incorrect behavior of RemoveAlwaysTrueIfConditionRector
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-main      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.com/demo/8d8a04b4-7402-4bb6-8126-f484e690111c

```php
<?php

final class ChildClass extends VendorClass
{
    private bool $executed;
    
    public function run(): void
    {
        $this->executed = false;
        
        parent::run();
        
        if (!$this->executed) {
            echo 'not run';
        }
    }
    
    protected function execute(): void
    {
        $this->executed = true;
        
        parent::run();
    }
}

class VendorClass
{
    public function run()
    {
        $this->execute();
    }
    protected function execute()
    {
    }
}

(new ChildClass)->run();
```

### Responsible rules

* `RemoveAlwaysTrueIfConditionRector`

## Expected Behavior

Rector should skip it. Adding the return value of `: void` in the vendor `run` method fixes the issue.

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/8583
Fixes https://github.com/rectorphp/rector/issues/8413

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉 /cc @georgec-floyt

File: rules/DeadCode/Rector/If_/RemoveAlwaysTrueIfConditionRector.php [hunk 1]
```diff
@@ -149,27 +146,7 @@ private function shouldSkipPropertyFetch(Expr $expr): bool
...
-            }
-
-            $nativeProperty = $classReflection->getNativeProperty($propertyName);
-            if (! $nativeProperty->hasNativeType()) {
-                return true;
-            }
-        }
-
-        return false;
+        return $propertyFetches !== [];
```
Comment (samsonasik) [inline rules/DeadCode/Rector/If_/RemoveAlwaysTrueIfConditionRector.php]: skip property fetch on if condition seems safer 👍

**Comment by samsonasik** [comment]
@gruberroland this also fix https://github.com/rectorphp/rector/issues/8413 as ArrayDimFetch type is unreliable to be removed, related with phpstan bug https://github.com/phpstan/phpstan/issues/10640

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

**Comment by ghost** [comment]
Thank you @samsonasik!

**Pull request merged**